src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala

```diff
@@ -16,19 +16,23 @@
 
 package com.microsoft.hyperspace.index
 
-import org.apache.spark.sql.SparkSession
-import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
 import org.apache.spark.sql.types.{DataType, StructType}
 
+import com.microsoft.hyperspace.HyperspaceException
 import com.microsoft.hyperspace.actions.Constants
-import com.microsoft.hyperspace.index.serde.LogicalPlanSerDeUtils
 
 // IndexLogEntry-specific fingerprint to be temporarily used where fingerprint is not defined.
 case class NoOpFingerprint() {
   val kind: String = "NoOp"
   val properties: Map[String, String] = Map()
 }
 
+case class RelationMetadataEntry(
```

Comment:
Just `RelationMetadata`? Also, I think this should capture only the info to recreate the DF, meaning `files` don't need to be here.

Reply:
I had the same thought but incremental refresh (https://github.com/microsoft/hyperspace/issues/29) will require `files` as well. Do we want to tackle it later?

Reply:
`files` will still be stored in `Source.data` right? I think the current approach is duplicating the `files`.

Reply:
got it. you are right. Thanks @imback82

Reply:
Actually, I had a discussion with @rapoth as well and came up with the following:
```
case class Relation(
    rootPaths: Seq[String],
    data: Hdfs,
    dataSchemaJson: String,
    fileFormat: String)

object SparkPlan {
  case class Properties(
    relations: Seq[Relation], // To support indexed views which can contain multiple relations.
    rawPlan: String, // null for now (will be used when we have IR to serialize a plan)
    sql: String, // null for now (will be used when we capture SQL)
    fingerprint: LogicalPlanFingerprint)
}

case class Source(plan: SparkPlan)
```

WDYT @sezruby?

Reply:
Looks good and it's a good idea to define new fields in advance 😀